### PR TITLE
Feat/search box events

### PR DESCRIPTION
**Pull request by iam4x**
**What project are you opening a pull request for?**
- react-instantsearch (use v2 base)

**Summary**

#2017

### CODE REVIEW

File: packages/react-instantsearch/src/components/SearchBox.js [hunk 1]
```diff
@@ -134,11 +136,19 @@ class SearchBox extends Component {
 
   onChange = e => {
     this.setQuery(e.target.value);
+
+    if (this.props.onChange) {
+      this.props.onChange(e);
```
Comment (Haroenv) [inline packages/react-instantsearch/src/components/SearchBox.js]: I usually use `addEventListener('input')`, is this the same in react?

Comment (iam4x) [inline packages/react-instantsearch/src/components/SearchBox.js]: Yes, but React automatically dispose listeners when the DOM node is removed. If you do this by your own you will need to `removeEventListener` but its not recommended, let React do the things 👍

**Comment by iam4x** [comment]
@mthuret Will do both, I have an idea for the story :+1:

**Comment by iam4x** [comment]
Here is the story @mthuret 



How do you organize your files into `stories` directory? Since I had to write more logic than expected, it feels a bit dirty to put everything into `stories/SearchBox.stories.js`

**Comment by mthuret** [comment]
Yes, by using the `action` function of storybook, you'll simplify a lot the story :)

File: packages/react-instantsearch/src/widgets/SearchBox.js [hunk 2]
```diff
@@ -8,6 +8,11 @@ import SearchBoxComponent from '../components/SearchBox.js';
  * @propType {string[]} [focusShortcuts=['s','/']] - List of keyboard shortcuts that focus the search box. Accepts key names and key codes.
  * @propType {boolean} [autoFocus=false] - Should the search box be focused on render?
  * @propType {boolean} [searchAsYouType=true] - Should we search on every change to the query? If you disable this option, new searches will only be triggered by clicking the search button or by pressing the enter key while the search box is focused.
+ * @propType {function} [onSubmit] - Intercept submit event sent from the SearchBox form container.
+ * @propType {function} [onReset] - Listen to `reset` event sent from the SearchBox form container.
```
Comment (vvo) [inline packages/react-instantsearch/src/widgets/SearchBox.js]: miss: [on*] - Will be forwarded to the actual used `input` allowing you to listen for keystrokes for examples.

File: stories/SearchBox.stories.js [hunk 3]
```diff
@@ -43,3 +43,73 @@ stories.add('default', () =>
...
+                  checked={ this.state.selectedEvents[eventName] }
+                  onChange={ this.handleSelectedEvent(eventName) }
+                />
+                { eventName }
+              </label>
+          ) }
+        </div>
+
+        <SearchBox
+          onSubmit={ this.handleSubmit }
```
Comment (vvo) [inline stories/SearchBox.stories.js]: Nice! Storybook FTW.

File: stories/SearchBox.stories.js [hunk 4]
```diff
@@ -43,3 +43,73 @@ stories.add('default', () =>
...
+    // we dont want to log unselected event
+    if (this.state.selectedEvents[eventName]) {
+      action(eventName)(event);
+    }
+  }
+
+  render() {
+    return (
+      <WrapWithHits searchBox={ false } hasPlayground={ true } linkedStoryGroup="searchBox">
+        <div style={ {color: '#999', borderBottom: '1px solid #E4E4E4', marginBottom: 10} }>
```
Comment (vvo) [inline stories/SearchBox.stories.js]: Maybe add a small text that says "look at the action logger in the sidebar"?

**Pull request merged**